datafusion/doc-gen/Cargo.toml

```diff
@@ -0,0 +1,35 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+[package]
+name = "datafusion-doc-gen"
```

Comment:
Since this holds `Documentation`, it seems like a name like `datafusion-doc` might be a name with more clarity

It would also likely help to explain why this needs to be in its own crate (so the macros can use it?)

Reply:
Yes, procedural macros crates are not very flexible and requires all structures to be predefined before the macros is used. Renaming it